docs: create documentation templates (#824)
rx-angular/rx-angular: docs/templates/PACKAGE_MAIN_DOCUMENTATION.MD

> **📦 Resources**

Reviewer: ```suggestion
# [FEATURE_NAME]

**📦 Resources**
```

Reviewer: Overall, maybe rename file to PACKAGE_FEATURE_DOCUMENTATION?

<!-- RESOURCES:
* required

Listing of resources related to the topic including running versions of used code snippets or other applications. 

> Resource links are prefixed with the "🔗" icon

Reviewer 2: are those internal or external resources?

Author: It was meant to link to the demo repo. I started to create a demo repo per package.

-->

Example:
> [🔗 GitHub](https://github.com/BioPhoton/rx-angular-cdk-coercing).

# ⛽ Motivation

<!-- MOTIVATION:
* required

Motivation is here to point out the value for the consumer. If possible we can make clear in which situations the impact will get visible most.

> Importent to cover in this section is:
> - Theoretical background to understand the problem
> - The problem and pain points in detail
> - Sketch a optimal solution for the problem

Reviewer 2: ```suggestion
This section should cover at least these main points:
- Problem description. What are existing pain points?
- Theoretical background to understand the problem
- Description of different solution approaches
- Sketch the optimal solution for the problem and explain why (should be our solution then :1st_place_medal:)
```

Reviewer 2: Oh I have just seen there is a dedicated chapter for different approaches. Do we really need to dedicate a whole section to it?

-->

Example:
> 

> # Available Approaches

Reviewer 2: ```suggestion
# Other Solution Approaches
```

<!-- AVAILABLE_APPROACHES:
* required

This section should sum up current available approaches and point out the mors importat differences.
The listing can be done in a bullet point list with optional links.
-->

Example:
> - Disabling zone.js completely (compiler options)
> - Disabling specific API's (zone configuration)
> - Overriding Angular dependencies (custom event manager)

> > - Using `unpatch`

Reviewer 2: If we stick to that section I would at least swap it with the RxAngular solution. This should be the first people get to see.

Reviewer 2: my point of view: I don't want users to scroll a lot until they finally see the RxAngular solution. It is most important for the doc, so it should be at the most prominent position

# 🎯 RxAngular Solution

## Installation and Setup

<!-- INSTALL_INSTRUCTIONS:
* required

Install istructions and code snippets/terminal commands to run. 
-->

Example:
> ```bash
> npm install --save @rx-angular/cdk
> # or
> yarn add @rx-angular/cdk
> ```

## Usage

<!-- USAGE:
* required

[TODO]
-->

## Best Practices

<!-- BEST_PRACTICES:
* required

This section should list the most important things to know when using the package. 

The following things could be covered:
- Good/Bad examples
- Folder and file organization
- Measurements

-->
----

## Testing

<!-- TESTING:

> * required

Reviewer 2: I wouldn't make this one required. I cannot imagine a dedicated testing section for  `coercing` or `coalescing`. The behavior itself is tested on our end. What do you want to explain here?
I see the point for the `template` or `state` package where one would have to take special care of providers or such things.

> [TODO]

Author: @edbzn it would be cool to get your feedback in here when you are done with the testing for `RxState`
